src/JustSaying/Messaging/Middleware/ErrorHandling/ErrorHandlerMiddlewareBuilderExtensions.cs

```diff
@@ -0,0 +1,17 @@
+using System;
+using JustSaying.Messaging.Monitoring;
+
+namespace JustSaying.Messaging.Middleware.ErrorHandling
+{
+    public static class ErrorHandlerMiddlewareBuilderExtensions
+    {
+        public static HandlerMiddlewareBuilder UseErrorHandler(this HandlerMiddlewareBuilder builder)
+        {
+            if (builder == null) throw new ArgumentNullException(nameof(builder));
+
+            IMessageMonitor monitor = builder.ServiceResolver.ResolveService<IMessageMonitor>();
```

Comment:
We can't just resolve the `ErrorHandlerMiddleware` from the container, because if the `IMessageMonitor` was added via the ServicesBuilder, then resolving the middleware from the container will bypass the SB and resolve from the container, which is a no-op by default.

The downside is that middlewares that do this are instance-per-queue instead of shared.